### PR TITLE
Naming conventions

### PR DESCRIPTION
**Pull request by ryexley**
* Added a new folder for naming conventions
* Created a new document with recommendation for server naming conventions

### CODE REVIEW

**Comment by cromwellryan** [comment]
I love this! I'm torn between this being in `naming_conventions` versus a `hosting` section. Thoughts?

**Comment by ryexley** [comment]
I'm not opposed to moving it to a `hosting` folder, though I chose this particular folder with the idea that we will (hopefully), eventually have _other_ naming conventions documents in there for things like 1Password entries, etc, etc. I'd be fine with either.

**Comment by cromwellryan** [comment]
We could cross link a hosting section to the appropriate naming conventions...

File: naming_conventions/servers.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
+Server Naming Conventions
+==========================
+
+Servers should be named consistently and in a manner in which they can be easily identifiable with regard to the client/property and purpose that they serve. We recommend the following naming conventions when creating/building new servers:
```
Comment (cromwellryan) [inline naming_conventions/servers.md]: I wonder if we should explicitly mention this is for Rackspace. On AWS we have been ignoring server name and instead relying on Role and Env tags. ¯\_(ツ)_/¯

For example:
> When using services like Rackspace for hosting, servers...

**Comment by ryexley** [comment]
I can get down with that. I'll modify the PR accordingly.

File: naming_conventions/servers.md (hunk 1 above)
Comment (ryexley) [reply naming_conventions/servers.md]: I wondered about that...but chose to be conservative in how much information we would want to share in a public repo. Maybe something a little more vague like

> When using services that support defining server/service names specifically...

And also adding a section for AWS conventions specifically? I dunno...not sure if my concern about specific services is unfounded and unnecessary...if so, yeah, I'd be cool with this.

I would only say that if we do something different for AWS, I feel like we should document that somewhere as well.

Comment (cromwellryan) [reply naming_conventions/servers.md]: I like your rendition, but I do want to include somewhere our preferred tooling.

Comment (ryexley) [reply naming_conventions/servers.md]: Perhaps in the new `hosting/README.md` that I'm creating for cross-linking to the naming conventions doc?

Comment (cromwellryan) [reply naming_conventions/servers.md]: Sure!

**Comment by ryexley** [comment]
PR updated with a new `hosting` folder with a `README.md` that cross-links to the server naming conventions document.

**Comment by cromwellryan** [comment]
Content looks good for now!

Couple recommendations regarding the commit messages themselves:

image

## 753979a
A commit provides a diff. Iterating the tactical actions that took place in the body of a commit doesn't add any new context to the commit for the reader. For example, we can see in the commit diff that you _Added a new folder for naming conventions_. Reiterating that in the body of the message isn't necessary. 

Instead, consider explaining the reason or mindset for creating this folder and the intent behind it. Someone reading this in the future is in the "diagnostics" mindset. "What's this commit about." "Why did we do or change this". 

This is why we recommend or prefer the [imperative tone](http://365git.tumblr.com/post/3308646748/writing-git-commit-messages) in [The Art of the Commit message](https://github.com/sparkbox/standard/tree/master/code-style/git#the-art-of-the-commit-message).

## f2e3a01
Consider splitting the [title and the body](https://github.com/sparkbox/standard/tree/master/code-style/git#the-art-of-the-commit-message) so that the commit title can be read in full. Right now it's too long to be seen on GH without the ellipses.

**Comment by ryexley** [comment]
Happy to update those commits accordingly. Would it be preferable in this case to just squash them into one in that format?

**Comment by cromwellryan** [comment]
totes!

**Comment by ryexley** [comment]
OK, the commit message has been updated, hopefully more appropriately.

**Pull request merged**